### ISSUE TITLE
Authenticate to AWS using profiles

### ISSUE THREAD

**Issue by znb** [turbot/steampipe-mod-aws-compliance#228]
**Is your feature request related to a problem? Please describe.**
We use SSO to authenticate against our AWS infrastructure. We have access via STS profiles which work for the AWS CLI tools but they don't appear to work for the compliance mods.

**Describe the solution you'd like**
It would be useful to be able to run the various compliance checks against our AWS accounts

**Additional context**
This is the error I get when I run this command to attempt and audit

```
ERROR: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
```

This is the command I ran:

```
AWS_PROFILE=productionaudit steampipe check benchmark.cis_v140
```

**Comment by cbruno10** [comment]
Hey @znb , sorry you are currently running into issues, hopefully we can figure out a solution.

How do you currently authenticate into the AWS CLI, e.g., do you generate an access key, secret key, and session token, and store them into env variables or your AWS credentials file manually? Or do you use the CLI interactively to login which then automatically retrieves and stores the creds for you?

Are you also able to provide what your AWS credentials + config files look like?

**Comment by znb** [comment]
We use Google SSO for identities and [this little CLI tool](https://github.com/cevoaustralia/aws-google-auth) to authenticate.
We then use the normal role switch to gain access to other accounts.

The high level setup is very similar to [this](https://www.marcolancini.it/2019/blog-cross-account-auditing/)

It all boils down to zero access keys and AWS STS for short term credentials. 

This all works fine with standard steampipe but authentication seems to fail when running the compliance mods.

**Comment by cbruno10** [comment]
@znb Thanks for the info!

Based on your latest comment, does that mean you're able to run something like `steampipe query "select name from aws_s3_bucket;"` for a given account and you'll get the correct results?

I'm not familiar with the tool you had linked, would you be able to share a log of the sequence you'd normally run, e.g., authenticating with aws-google-auth and then running the Steampipe command(s)? Also, are you able to share your `~/.steampipe/config/aws.spc` configuration?

**Comment by znb** [comment]
Yes, if I load up `steampipe query` and run my queries, they work as expected.

However if I authenticate and get a temporary set of credentials, I then run `steampipe check all` (as an example) and get the above error.

Here's an example of my `~/.steampipe/config/aws.spc`

```
connection “infosec” {
  plugin      = “aws”
  profile     = “infosec-audit”
  regions     = [“eu-west-1"]
}
```

Where "infosec-audit" is an AWS profile configured in `~/.aws/config`

I think the main issue is that either the credentials aren't getting passed to steampipe correctly or something else isn't working as expected.

Hope this is more useful

**Comment by znb** [comment]
I can confirm it's definitely a problem with the `aws-google-auth` credentials. On a second account I have a role which I can access via an IAM role switch with access keys.

A simple `AWS_PROFILE=infosec steampipe check all` runs the checks as expected in the correct account.

I'll investigate a little more in the morning. If there's anything I can do to bump up the logging to find out where the authentication is failing, please let me know.

I'm also happy to close off this ticket as I suspect I'd be a weird edge case for the most part 😄
